op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,363 @@
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/gorilla/websocket"
+)
+
+// FlashblockHandler manages WebSocket connections for flashblocks
+type FlashblockHandler interface {
+	// Start initializes and starts the flashblocks handler
+	Start(ctx context.Context) error
+	// Stop closes any open WebSocket connections
+	Stop()
+	// BroadcastMessage sends a message to all connected WebSocket clients
+	BroadcastMessage(message []byte)
+}
+
+// Config contains configuration for the flashblocks handler
+type Config struct {
+	// WebsocketServerPort is the port to listen for WebSocket connections
+	WebsocketServerPort int
+	// RollupBoostWsURL is the URL of the rollup boost WebSocket
+	RollupBoostWsURL string
+}
+
+// Client represents a connected WebSocket client
+type Client struct {
+	conn         *websocket.Conn
+	send         chan []byte
+	failureCount int
+	maxFailures  int
+}
+
+// Close closes the client's WebSocket connection and send channel
+func (c *Client) Close() {
+	close(c.send)
+	c.conn.Close()
+}
+
+// newClient creates a new client with default settings
+func newClient(conn *websocket.Conn) *Client {
+	return &Client{
+		conn:        conn,
+		send:        make(chan []byte, 1), // Buffer size of 1
+		maxFailures: 5,                    // Drop client after 5 consecutive failures
+	}
+}
+
+// Hub maintains the set of active clients and broadcasts messages to them
+type Hub struct {
+	// Registered clients
+	clients map[*Client]bool
+	// Mutex to protect access to clients map
+	clientsMu sync.RWMutex
+
+	// Register requests from the clients
+	register chan *Client
+
+	// Unregister requests from clients
+	unregister chan *Client
+
+	// Inbound messages to broadcast to the clients
+	broadcast chan []byte
+
+	// Signal to stop the hub
+	done chan struct{}
+}
+
+// newHub creates a new hub
+func newHub() *Hub {
+	return &Hub{
+		broadcast:  make(chan []byte, 256),
+		register:   make(chan *Client),
+		unregister: make(chan *Client),
+		clients:    make(map[*Client]bool),
+		done:       make(chan struct{}),
+	}
+}
+
+// run starts the hub's main loop
+func (h *Hub) run() {
+	for {
+		select {
+		case <-h.done:
+			// Close all client connections
+			h.clientsMu.Lock()
+			for client := range h.clients {
+				client.Close()
+				delete(h.clients, client)
+			}
+			h.clientsMu.Unlock()
+			return
+		case client := <-h.register:
+			h.clientsMu.Lock()
+			h.clients[client] = true
+			h.clientsMu.Unlock()
+		case client := <-h.unregister:
+			h.clientsMu.Lock()
+			if _, ok := h.clients[client]; ok {
+				delete(h.clients, client)
+				client.Close()
+			}
+			h.clientsMu.Unlock()
+		case message := <-h.broadcast:
+			h.clientsMu.RLock()
+			for client := range h.clients {
+				select {
+				case client.send <- message:
+					// Message sent successfully
+					client.failureCount = 0 // Reset failure count on success
+				default:
+					// Client send buffer is full, increment failure count
+					client.failureCount++
```

Comment:
send out unregister signal here instead of doing it directly in broadcast

1. better for broadcast performance
2. avoid all the tricky lock / unlock situations
3. also probably change unregister to be a buffered channel as well to handle multiple events